certchain/certmanager/augmentor.go

```diff
@@ -107,9 +107,6 @@ func (a *Augmentor) Start() error {
 		return err
 	}
 	orNext, err := a.maintainOCSPResp()
-	if err != nil {
```

Comment:
Yusuke, I had to remove this check and return statement.  In the event that during start-up, we had an expired certificate and ACME retrieved it, it is very common for DigiCert OCSP response to fail for a period of time (due to updates of their data in the CDN).  In this case, it will be beneficial to retry until we get the proper OCSP response.  In the previous state of the code, no retry was attempted and the server exits with an error.

Reply:
I understand there is a problem but I'm not fully sure if you _had_ to ― I don't think dropping this if statement is the only option. At the very least the code is now violating the contract written in GoDoc: "It produces the first AugmentedChain before starting the goroutine and blocks until it is ready." This change can introduce subtle bugs since the rest of codebase is written with relying on that contract.

I have to say removing this check is over-focused on the DigiCert+ACME case. With this change Augmentor won't be able to handle some bad certificates in a proper way, including:

* those truly revoked, for which the OCSP responder will never return okay.
* those with a bogus OCSP URI, which can't be accessed in the first place.

The most straightforward approach is to do a "white-true" loop to wait until the function receives a valid ocsp response:

```go
var orNext futureevent.Event
for {
  orNext,err = a.maintainOCSPResp()
  if err == nil {
    break
  }
  log.Print(err)
  <-orNext.Chan()  // I don't mean a busy loop.
}
```

If possible, it's nicer to determine whether the failure is transient or permanent, and exit the loop on a permanent error.

Reply:
Thanks for the comments. Updated to your suggested logic but added a retry counter so that it can break out of the loop after N retries.